### ISSUE TITLE
External avatar urls broken in topposter-block

### ISSUE THREAD

**Issue by GWRon** [Tinyportal/TinyPortal#966]
Hey,

I am not a direct user of the tinyportal plugin - but a user of a forum utilizing it.

There I recognized that "SMF" allows configuring external avatar urls - as seen in the shot below:
![image](https://github.com/Tinyportal/TinyPortal/assets/2625226/6e9da4a0-3721-47cb-8cc5-54b04eb7b8e8)

But in your tinyportal topposter-block-code you only handle internal avatar urls or urls with "http://":

tp-files/tp-blockcodes/topposters.blockcode:
```PHP
'image' => empty($row['avatar']) ? ($row['id_attach'] > 0 ? 'src="' . (empty($row['attachment_type']) ? $scripturl . '?action=dlattach;attach=' . $row['id_attach'] . ';type=avatar' : $modSettings['custom_avatar_url'] . '/' . $row['filename']) . '" alt="" style="'.$style.'" title="'.$row['real_name'].'" />' : '') : (stristr($row['avatar'], 'http://') ? 'src="' . $row['avatar'] . '" alt="" style="'.$style.'"  title="'.$row['real_name'].'" />' : 'src="' . $modSettings['avatar_url'] . '/' . htmlspecialchars($row['avatar']) . '" alt="" style="'.$style.'" title="'.$row['real_name'].'" />'),
```
(code from latest change/commit: https://github.com/Tinyportal/TinyPortal/commit/97a8a398f5e33b0720fb3ee0e7ae73e48de69dcf)

As seen in above code snippet you simply concat two strings regardless of `$row['avatar']` containing an external url with httpS.

To emphasize it - this is a bit reformatted for easier readability what the terniary operators do:
```
'image' => 
	empty($row['avatar'])
	? ($row['id_attach'] > 0 
		? 'src="' . (empty($row['attachment_type']) 
					? $scripturl . '?action=dlattach;attach=' . $row['id_attach'] . ';type=avatar' 
					: $modSettings['custom_avatar_url'] . '/' . $row['filename']
				). '" />' 
		: '') 
	: 
		(stristr($row['avatar'], 'http://') 
		? 'src="' . $row['avatar'] . '" />' 
		: 'src="' . $modSettings['avatar_url'] . '/' . htmlspecialchars($row['avatar']) . '"  />'
		),
```

----

A simple fix would be to check for "http://" OR "https://" there. Regarding "security" I consider whatever is stored in `$row['avatar']` is already sanitized/checked/... by the SMF core code (so profile editor).

Maybe replacing `(stristr($row['avatar'], 'http://')` with `(stripos($row['avatar'], 'http://') == 0 or stripos($row['avatar'], 'https://') == 0)` (or use a proper regex) will be good enough.

**Comment by tinoest** [comment]
Thanks for raising this.

We would be better changing that block to use the set_avatar_data function which is available in SMF2.1 as that would offer everything SMF does

**Comment by Bopske** [comment]
Created new blockcode for use in SMF2.1. This is included in version 2.3.1 of Tinyportal
